### PR TITLE
Spreadsheet styling, backend refactoring

### PR DESCRIPTION
**Pull request by benjamin-shen**
### Summary 

This pull request adds finishing touches to the Export functionality of Flux Fitness.

- [x] implemented spreadsheet styling
- frozen rows/columns
- cell alignment
- newlines for granular data
- tab coloring
- auto cell height/width
- [x] refactored main function in `index.ts`

### Test Plan 

The cloud function is deployed. Run `npm run serve` to locally test the export functionality on the web app. The downloaded spreadsheet should have different formatting but display the same information.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "axios": "^0.19.2",
     "blob": "0.0.5",
     "core-js": "^3.6.5",
+    "exceljs": "^3.9.0",
```
Comment (kungpaogao) [inline package.json:15]: Is this a mistake?

**Comment by kungpaogao** [comment]
yikes, I made an oopsy. I have to check this out locally.

File: functions/src/index.ts [hunk 2]
```diff
@@ -206,3 +107,172 @@ function roundDate(d: Date) {
...
+      const latestTime = latest.hour() * 60 + latest.minute();
+      endTime = Math.max(endTime, latestTime);
+      data = dateData;
+    }
+    separateDates.push(data);
+  }
+  return [beginTime, endTime];
+}
+
+function generateDates(startDate: any, endDate: any) {
```
Comment (kungpaogao) [inline functions/src/index.ts:137]: could probably import the `Moment` type for these

File: functions/src/index.ts [hunk 3]
```diff
@@ -206,3 +107,172 @@ function roundDate(d: Date) {
...
+      treadmills +
+      " \nEllipticals: " +
+      ellipticals +
+      " \nBikes: " +
+      bikes +
+      " \nAMTs: " +
+      amts;
+    const cardioTotalCell = treadmills + ellipticals + bikes + amts;
+    cardioGranularData = cardioGranularCell;
+    cardioTotalData = cardioTotalCell;
```
Comment (kungpaogao) [inline functions/src/index.ts:189]: Is there a reason why you create a new `const` instead of assigning directly to `cardioGranularData`?

Comment (benjamin-shen) [reply functions/src/index.ts:189]: Hmm not that I can remember, I was probably preparing for more manipulation. I'll simplify it.

**Pull request merged**